src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,90 @@
+package frc.robot.subsystems;
+
+import static edu.wpi.first.units.Units.Rotations;
+import static edu.wpi.first.units.Units.RotationsPerSecond;
+import static edu.wpi.first.units.Units.RotationsPerSecondPerSecond;
+
+import com.revrobotics.spark.SparkAbsoluteEncoder;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkFlex;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import edu.wpi.first.math.controller.ArmFeedforward;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Configs;
+import frc.robot.Constants.ArmConstants;
+import frc.robot.Constants.ArmSetpoint;
+
+public class ArmSubsystem extends SubsystemBase {
+  // Arm Motor
+  private final SparkFlex armMotor = new SparkFlex(ArmConstants.ARM_MOTOR, MotorType.kBrushless);
+  private final SparkAbsoluteEncoder armEncoder = armMotor.getAbsoluteEncoder();
+
+  private final ProfiledPIDController armPid =
+      new ProfiledPIDController(
+          ArmConstants.kP,
+          ArmConstants.kI,
+          ArmConstants.kD,
+          new TrapezoidProfile.Constraints(
+              (ArmConstants.ARM_MAX_VELOCITY).in(RotationsPerSecond),
+              (ArmConstants.ARM_MAX_ACCELERATION).in(RotationsPerSecondPerSecond)));
+
+  private final ArmFeedforward armFeedforward =
+      new ArmFeedforward(ArmConstants.kS, ArmConstants.kG, ArmConstants.kV, ArmConstants.kA);
+
+  // Default Current Target
+  private ArmSetpoint armSetpoint = ArmSetpoint.START;
+
+  /** Creates a new ArmSubsystem. */
+  public ArmSubsystem() {
+    armMotor.configure(
+        Configs.ArmConfig.ARM_CONFIG,
+        ResetMode.kResetSafeParameters,
+        PersistMode.kPersistParameters);
+  }
+
+  public boolean atSetpoint(ArmSetpoint setpoint) {
+    boolean a = armSetpoint == setpoint;
+    boolean b =
+        Math.abs(armEncoder.getPosition() - setpoint.value.in(Rotations))
+            <= ArmConstants.ARM_TOLERANCE.in(Rotations);
+    return a || b;
+  }
```

Comment:
You should have two methods here:
```suggestion
public boolean atSetpoint() {
    return
        Math.abs(armEncoder.getPosition() - armSetpoint.value.in(Rotations))
            <= ArmConstants.ARM_TOLERANCE.in(Rotations);
  }
  public boolean atSetpoint(ArmSetpoint setpoint) {
    boolean a = armSetpoint == setpoint;
    boolean b =
        Math.abs(armEncoder.getPosition() - setpoint.value.in(Rotations))
            <= ArmConstants.ARM_TOLERANCE.in(Rotations);
    return a || b;
  }
```

You should be using the former to make sure that you have reached it. Honestly, I think the `setSetpoint` should have a `.waitUntil(atSetpoint);`. The latter should be used if you're checking that the arm is not in a certain state. You can potentially remove the at `a` condition too.

Reply:
Can I just keep the former method? Because if I just remove boolean a, it will run the exact same as the former, except with the other method you're choosing what the method runs with what setpoint.

Reply:
Okay sure, if you think that's simpler